aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -125,17 +131,144 @@ macro_rules! unwrap_or_discard {
     };
 }
 
-pub(crate) fn discard_error_vm_status(vm_status: VMStatus) -> (VMStatus, VMOutput) {
+fn discard_error_vm_status(vm_status: VMStatus) -> (VMStatus, VMOutput) {
     let vm_output =
         VMOutput::empty_with_status(TransactionStatus::Discard(vm_status.status_code()));
     (vm_status, vm_output)
 }
 
+fn get_transaction_output(
+    session: SessionExt,
+    fee_statement: FeeStatement,
+    status: ExecutionStatus,
+    change_set_configs: &ChangeSetConfigs,
+) -> Result<VMOutput, VMStatus> {
+    let change_set = session.finish(change_set_configs)?;
+    Ok(VMOutput::new(
+        change_set,
+        fee_statement,
+        TransactionStatus::Keep(status),
+    ))
+}
+
+pub(crate) fn gas_config(
+    config_storage: &impl ConfigStorage,
+) -> (Result<AptosGasParameters, String>, u64) {
```

Comment:
Could also simplify the implementation to smt like:
```
let maybe_schedule_and_version = GasScheduleV2::fetch_config(config_storage).map(|gas_schedule| (gas_schedule, gas_schedule.feature_version)).or(GasSchedule::fetch_config(config_storage).map(|gas_schedule| (gas_schedule, 0)));

maybe_schedule_and_version.map(|(gas_schedule, feature_version)| {
            let map = gas_schedule.to_btree_map();
            (
                AptosGasParameters::from_on_chain_gas_schedule(&map, feature_version),
                feature_version,
            )
}).ok_or("Neither gas schedule v2 nor v1 exists.".to_string())
```

although not sure it's that much "simpler" xD

Reply:
Indeed, the problem here though is that we store Result<AptosGasParameters> and Result<StorageGasParamters> inside the VM :( And these error only when called, not at the real VM creation time. I tried to rewrite that but this adds quite a bit of changes because VM creation can fail now and this has implications on simulation, view function interfaces and block-execution. Let's do that in the next PR? Otherwise 100% we should do that